app/assets/javascripts/float-kit/addon/modifiers/close-on-click-outside.js

```diff
@@ -0,0 +1,33 @@
+import Modifier from "ember-modifier";
+import { registerDestructor } from "@ember/destroyable";
+import { bind } from "discourse-common/utils/decorators";
+
+export default class FloatKitCloseOnClickOutside extends Modifier {
+  constructor(owner, args) {
+    super(owner, args);
+    registerDestructor(this, (instance) => instance.cleanup());
+  }
+
+  modify(element, [trigger, closeFn]) {
+    this.closeFn = closeFn;
+    this.trigger = trigger;
+    this.element = element;
+    window.addEventListener("click", this.check, {
+      passive: true,
+    });
```

Comment:
I know that there is probably not another way around this, but event listeners on `window` gives me anxiety.

Reply:
why?

Reply:
It's kinda low risk here, as it's passive and has limited actions

Reply:
window is such a massive scope, that is all.

Reply:
I can move it to document, I see this is what tippy is doing for example, that's not going to change much though :+1: